### PR TITLE
Add spec validation

### PR DESCRIPTION
**Pull request by Setogit**
connect to https://github.com/strongloop/loopback-next/issues/495

### CODE REVIEW

File: src/controllers/hello-world.ts [hunk 1]
```diff
@@ -1,7 +1,11 @@
 import {api} from '@loopback/core';
+import {createEmptyApiSpec, OpenApiSpec} from '@loopback/openapi-spec';
 import {apiSpec} from './hello-world.api';
 
-@api(apiSpec)
+var openApiSpec: OpenApiSpec = createEmptyApiSpec();
+Object.assign(openApiSpec, apiSpec);
```
Comment (bajtos) [inline src/controllers/hello-world.ts]: Why is this needed? Is some code modifying our OpenApiSpec object? IIRC, we are always creating a clone of the spec objects to avoid the need to copy them. 

If your goal is to add fields from empty api spec, then please note that `@api` decorator no longer requires a full Swagger preamble, it takes a smaller object with two properties only: `{basePath?, paths}`.

File: test/spec.smoke.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+const http = require('http');
+const Dredd = require('dredd');
+const port = 3030;
+const localhost = 'http://localhost:' + port;
+const dreddConfig = {
+  server: localhost + '/', // base path to the end points
+  options: {
+    silent: true,
+    path: [localhost + '/openapi.json'], // to download our apiSpec
```
Comment (bajtos) [inline test/spec.smoke.js]: Can we use the spec returned by `app.getApiSpec()`?

File: test/spec.smoke.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  function initEnvironment(config) {
+    app = createApp();
+    apiSpec = app.getApiSpec();
+    server = startApp(app);
+    dredd = new Dredd(config);
+  }
+
+  function startApp(app) {
+    const server = http.createServer(app.handleHttp);
+    server.listen(port, (err) => {
```
Comment (bajtos) [inline test/spec.smoke.js]: Please consider using `app.start()` instead.

File: test/spec.smoke.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
+const testlab = require('@loopback/testlab');
+const createApp = require('../dist/application').createApp;
+const expect = testlab.expect;
+const validateApiSpec = testlab.validateApiSpec;
+
+const http = require('http');
+const Dredd = require('dredd');
+const port = 3030;
```
Comment (bajtos) [inline test/spec.smoke.js]: Please use ephemeral port `0`, otherwise this test will fail when there is already some other process listening on port 3000, as may be the case in CI environments running multiple jobs in parallel on the same machine.

**Comment by Setogit** [comment]
the spec validation part has been merged as part of another PR: #10 
after that, the spec testing part was added on top it and PR'ed as #13 
comments above are incorporated accordingly.  thanks!
closing.
